src/gtc/passes/gtir_pipeline.py

```diff
@@ -31,9 +32,15 @@ class GtirPipeline:
     May only call existing passes and may not contain any pass logic itself.
     """
 
-    def __init__(self, node: gtir.Stencil):
+    _cache: Dict[Tuple[StencilID, Tuple[PASS_T, ...]], gtir.Stencil] = {}
```

Comment:
Can you help me understand this change? It looks to me like a unique `GtirPipeline` object is created for each `StencilID` anyway, since one is made for each stencil.

Reply:
Indeed, with the usual `gtscript.stencil` decorator, there is never more than just one gtir pipeline, since subsequent calls to the decorator load and return the previously generated module from file cache. However, with the `lazy_stencil` this is not the case: There is no cache on disk, yet e.g. in check_syntax or when accessing field_info, this triggers a pipeline each time. This would cache those cases so that for the same stencil (by hash, not by instance) the pipeline is only executed once.